mono/metadata/assembly.c

```diff
@@ -1764,7 +1764,7 @@ mono_assembly_load_from_full (MonoImage *image, const char*fname,
 	 * assemblies lock.
 	 */
 	if (ass->aname.name) {
-		ass2 = mono_assembly_invoke_search_hook_internal (&ass->aname, refonly, FALSE);
+		ass2 = mono_assembly_invoke_search_hook_internal (&ass->aname, NULL, refonly, FALSE);
```

Comment:
This (and the hunk below) doesn't seem right. `mono_assembly_load_from_full` is invoked from e.g. `ves_icall_System_AppDomain_LoadAssemblyRaw` which is used in `AppDomain.Load ()`. It's also used in many other API functions in `assembly.c` which should probably also have a `requesting` parameter added.